Navbar Redesign

Ron-Lavi opened the pull request:
In continuous to the Brno meetup redesign session @lzap @bastilian 
There are still some small things to fix but let's see if we like the nav behavior.   

![Foreman (1) (1)](https://user-images.githubusercontent.com/26363699/59566277-a1600800-9066-11e9-9793-37aff3e1cb8e.gif)

![Foreman __ Manual](https://user-images.githubusercontent.com/26363699/59566276-9efdae00-9066-11e9-8bfe-9807b98107ea.gif)

lzap commented:
Thanks @Laviro couple of comments:

* When I hover them quickly from left to right or vice versa they keep all opened. Is this something we can fix?
* Submenus are a bit off from main menu items, is this intentional? Does not look good.
* Only first items are clickable - looks like a bug?

Thanks for help with this one, this would took me hours of experimenting.

Ron-Lavi commented:
Thanks @lzap :)
I just uploaded a very early implementation to see if we even like it,
now that I know we do, I will dive in and fix all of these issues.

ronlavi2412 committed:
Add Navbar dropdowns

Ron-Lavi commented:
@lzap I just fixed the issues that you mentioned,
still need to find a fix to the navbar on small screens.

Ron-Lavi commented:
Added responsive dropdown to the navbar,
@lzap I think we are ready :)

![Foreman __ Foreman Blog](https://user-images.githubusercontent.com/26363699/59600076-df156d00-9108-11e9-98b0-822aadfba125.gif)

lzap commented on _config.yml:
```diff
@@ -27,17 +27,48 @@ markdown:    kramdown
 include:
 - _static
 
+# TODO use variables instead of 1.21
```
Drop this please, variables in config.yml cannot be used according to the documentation. We used to have it here anyway (see `1.20` on deleted like 34) and let's not try to solve this problem here.

lzap commented on _config.yml:
```diff
@@ -27,17 +27,48 @@ markdown:    kramdown
+      - title: Who uses Foreman
+        url: /who_uses.html
+      - title: Quickstart
+        url: /manuals/1.21/quickstart_guide.html
```
Can you change all occurances of 1.21 with 1.22 which was released in the meantime? :-)

ekohl replied:
We also have a latest redirect (which we should update to 1.22 now).
```suggestion
        url: /manuals/latest/quickstart_guide.html
```

lzap replied:
Oh nice please do it for all menu items then :-)

lzap reviewed (changes requested):
We are almost there, I'd appreciate comments from someone else preferably @tbrisker who often drives release process. Also @bastilian was working with us in Brno. Few comments:

1) Delete please page `documentation.md` because we have effectively moved all the links from there to the menu (yay!).
2) The last remaining bit is to create a simple javascript switcher on documentation page. Currently the only way to change manual version was via the dropdown on the documentation page (https://theforeman.org/documentation.html) and we need to create the very same but on the documentation page itself - ideally on the top left side above the topic menu (https://theforeman.org/manuals/1.22/index.html).
3) Optionally we could implement the same selector for Katello documentation (it has its own selector on the right side) and plugins.
4) There is a missing page Who uses foreman and I am going to provide you the missing markdown file - will send it to you.
5) Menu works nice in Chrome but in Firefox it does not - it's shifted to the left :-(

lzap commented:
Here is the missing file `who_uses_foreman.md` please put it there:

```markdown
---
layout: default
title: Who Uses Foreman
---

The following companies or individuals have confirmed use of Foreman. If you 
want help to spread the word, [let us
know](https://docs.google.com/a/zapletalovi.com/forms/d/1Ybz-Wz-odbGJM6Kw1pZuIOsZZ_vxXbH1I8B-iKFfuPs)
and we will add you to the page.

The list is in alphabetical order.

* [Brandorr Group](https://www.brandorr.com)
* [CERN](https://home.cern)
* [Linköpings universitet](https://liu.se)
* [Paychex](https://www.paychex.com)
* [Sixt](https://www.sixt.de)
* [ThinkIT Data Solutions](http://thinkitdata.com/)
* [Tipsport.net](https://www.tipsport.cz)
```

ekohl commented on _config.yml:
```diff
@@ -27,17 +27,48 @@ markdown:    kramdown
+  - title: Get Help
+    sublinks:
+      - title: Manual
+        url: /manuals/1.21/index.html
```
```suggestion
        url: /manuals/latest/index.html
```

ekohl commented on _config.yml:
```diff
@@ -27,17 +27,48 @@ markdown:    kramdown
+      - title: Plugins
+        url: /plugins/
+      - title: API docs
+        url: /api/1.21/index.html
```
```suggestion
        url: /api/latest/index.html
```

ekohl commented on _config.yml:
```diff
@@ -27,17 +27,48 @@ markdown:    kramdown
+      - title: Blog
+        url: /blog/index.html
+      - title: Events
+        url: https://theforeman.org/events
```
Should this be relative?

```suggestion
        url: /events
```

Ron-Lavi replied:
seems like these urls are coming from the `_layouts` folder,
while the `events` are redirecting to `https://community.theforeman.org/c/events/l/calendar`

@lzap may know better

lzap replied:
I have no idea, let's keep it as is.

ronlavi2412 committed:
refactor urls to 'latest'

ronlavi2412 committed:
Remove documentation.md

ronlavi2412 committed:
Add who_uses page

Ron-Lavi commented:
Thanks @ekohl @lzap :)

summary of the last commits:
- [x] Fixed the `firefox` `fit-content` problem.
- [x] Updated the urls in `_config.yml`.
- [x] Deleted `documentation.md`.
- [x] Added `Who Uses Foreman` page.
- [x] Need to create the version selector on the manual page

ronlavi2412 committed:
Added version selector to the "manual" page

Ron-Lavi commented:
@lzap, I just added the version selector on the "manual" page,
I added it above the header and not on the top left side above the topic menu,
since the page is responsive and on smaller screens the whole side is hidden:

![Foreman __ Manual (2)](https://user-images.githubusercontent.com/26363699/59680253-8eba1000-91d9-11e9-92bd-33d602c8baf9.gif)

lzap reviewed (approved):
It works just fine! Thanks so much, let's merge this and ask others for comments.

evgeni commented:
@Laviro @lzap this seems to have broken https://theforeman.org/api/latest/index.html (linked from "get help" → "api") and https://theforeman.org/documentation.html (linked from the "Latest versions: 1.22.0 / 1.21.3 / 1.20.2" in the front-page tease)

evgeni commented:
API should be fixed via https://github.com/theforeman/foreman-infra/pull/1082

evgeni commented:
and documentation.html should be fixed via https://github.com/theforeman/theforeman.org/pull/1399

lzap commented:
Thanks, the infra one was merged 20 hours ago however I am still getting 404. Any ideas?

evgeni commented:
@lzap yeah I think there was an issue with the puppetmaster and @ekohl was fixing it
